File: Modules/Common/Equipments/SIE.Web.Fixtures/Models/Commands/SelEquipModelCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.Fixtures.Models;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.Fixtures.Models.Commands
{
    /// <summary>
    /// 添加设备型号
    /// </summary>
    [JsCommand("SIE.Web.Fixtures.Models.Commands.SelEquipModelCommand")]
    public class SelEquipModelCommand : ViewCommand
    {
        /// <summary>
        /// 执行添加
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find(args.Type);
            var savedData = RF.Find(meta.EntityType).NewList();
            var equipDetails = args.Data.ToJsonObject<List<FixtureModelEquipDetail>>();
            Check.NotNullOrEmpty(equipDetails, nameof(equipDetails));
            if (null == equipDetails || equipDetails.Count == 0)
                throw new ValidationException("设备清单不能为空!".L10N());
            foreach (var item in equipDetails)
            {
                var equipDetail = new FixtureModelEquipDetail();
                equipDetail.FixtureModelId = item.FixtureModelId;
                equipDetail.EquipModelId = item.EquipModelId;
                savedData.Add(equipDetail);
            }
            RF.Save(savedData);
            return true;
        }
    }
}

File: Modules/Common/Equipments/SIE.Web.Fixtures/Models/Commands/SelEquipModelCommand.js
SIE.defineCommand('SIE.Web.Fixtures.Models.Commands.SelEquipModelCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'EquipModelId', targetClassName: 'SIE.Equipments.EquipModels.EquipModel' }
    },
    /**
     * canExecute 是否执行
     * @param {} view 当前视图
     * @returns {}
     */
    canExecute: function (view) {
        var entity = view.getParent().getCurrent();
        if (entity != null && entity.data) {
            return entity.data.INV_ORG_ID !== null;
        }
        return false;
    }, 
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    save: function (win) {
        /// <summary>
        /// 保存选择的操作列表。
        /// </summary>
        var me = this;
        var indata = {};
        var selections = me._targetSelectItems;
        if (selections && selections.items.length > 0) {
            var items = [];
            SIE.each(selections.items, function (sel) {
                var id = sel.getId();
                if (me._sourceViewSelectItems.indexOf(id) === -1) {
                    var item = { FixtureModelId: me._sourceId, EquipModelId: id };
                    items.push(item);
                }
            });
            if (items.length > 0) {
                indata = items;
                me._targetView.execute({
                    data: indata,
                    success: function (res) {
                        win.close();
                        me._ownerView.loadChildData(true);
                    }
                }, me._ownerView);
                return true;
            }
        }
        Ext.Msg.alert('提示'.t(), '没有可提交的数据'.t());
    }
});